### ISSUE TITLE
Android TV settings export Vestel 17MB170, Hitachi, Philips Telefunken

### ISSUE THREAD

**Issue by JP-Service** [PredatH0r/ChanSort#299]
Hallo. Please - I urgently need a way to export settings to USB. Concerning 2021/2022 Android Models of different brands. 
Examples:  Telefunke HH32AJ600, Hitachi U50KA6150 / H32E2200 or Philips 43PUS7906/12.
Thank You in advance, best regards. Jens

**Comment by PredatH0r** [comment]
Hi Jens. 
Sorry for the late response. The situation with Android TVs based on MediaTek chips (used by Vestel) is unfortunately grim.
My day job is not related to TVs or consumer electronics, so I don't have any reliable insights into the OEM market and making some assumptions here.

It appears to me that many TV brands buy hardware from a few board manufacturers, which in turn buy parts and firmware from even fewer companies. Android only covers aspects around the user interface and apps, but is agnostic of the low-level details around channel tuning and channel list imports/exports. The tuner part and software for it is probably provided by the chip manufacturer. So while many brands offer "Android" TVs, the actual tuning and channel list export/import depends on the chip and its firmware.

From what I have seen so far, Android TVs based on the Vestel platform use chips (and I assume also part of the firmware) from MediaTek. They all export channel list files with the same proprietary compressed/encrypted ".bin" format that I have seen on 10yr old Philips TVs before that also use MediaTek chips. Unfortunately nobody except MediaTek knows how to read or write these files.
In the past there was always a "readable" text file along side the encrypted .bin files that could be used to reorder the channels. But unfortunately the current MediaTek file format used in Android TVs produces a text file that is broken in so many ways that it can't be used for anything meaningful.

So far I received MediaTek based Android TV lists from Nokia, Xiaomi and Panasonic models.
The channel_list.xml file that comes with these TVs can be edited with ChanSort, but the file only holds 4-8 characters for each channel name and if multiple sources are in use (DVB-T, DVB-S), it doesn't specify to which list a channel belongs.
I do not have access to a MediaTek based Android TV and never received feedback, if the TV processes changes made to the .xml file when importing the list.

There are other brands with Android TVs that use an entirely different format, e.g. "Alden" (using a SQLite database file that can be edited with ChanSort) or TCL (also SQLite, but completely different structure, not yet supported by ChanSort).

Long story short: It would require MediaTek to fix the broken .xml file export and TV manufacturers to provide updates for the TV firmware - or MediaTek to disclose information about their proprietary .bin format (which is very unlikely to happen).
So either way it's up to MediaTek.

**Comment by PredatH0r** [comment]
Closing because there is nothing that can be fixed on the ChanSort side. 
Vestel or the manufacturer selling the TV under its brand will have to provide a usable non-proprietary unencrypted export format.
